utils/utils.go

```diff
@@ -284,23 +286,64 @@ func BuildTargetPath(pattern, path, target string, ignoreRepo bool) (string, boo
 
 	groups := r.FindStringSubmatch(path)
 	if len(groups) > 0 {
-		target, replaceOccurred := ReplacePlaceHolders(groups, target)
+		target, replaceOccurred, err := ReplacePlaceHolders(groups, target, isRegexp)
+		if err != nil {
+			return "", false, err
+		}
 		return target, replaceOccurred, nil
 	}
 	return target, false, nil
 }
 
-// group - regular expression matched group to replace with placeholders
-// toReplace - target pattern to replace
-// Return - (parsed placeholders string, placeholders were  replaced)
-func ReplacePlaceHolders(groups []string, toReplace string) (string, bool) {
+// ReplacePlaceHolders replace placeholders with their matching regular expressions.
+// group - Regular expression matched group to replace with placeholders.
+// toReplace - Target pattern to replace.
+// isRegexp - When using a regular expression, all parentheses content in the target will be at the given group parameter.
+// A non-regular expression will, however, allow us to consider the parentheses as literal characters.
+// The size of the group (containing the parentheses content) can be smaller than the maximum placeholder indexer - in this case, special treatment is required.
+// Example : pattern: (a)/(b)/(c), target: "target/{1}{3}" => '(a)' and '(c)' will be considered as placeholders, and '(b)' will be treated as the directory's actual name.
+// In this case, the index of '(c)' in the group is 2, but its placeholder indexer is 3.
+// Return - The parsed placeholders string, along with a boolean to indicate whether they have been replaced or not.
+func ReplacePlaceHolders(groups []string, toReplace string, isRegexp bool) (string, bool, error) {
+	maxPlaceholderIndex, err := getMaxPlaceholderIndex(toReplace)
+	if err != nil {
+		return "", false, err
+	}
 	preReplaced := toReplace
+	// Index for the placeholder number.
+	placeHolderIndexer := 1
 	for i := 1; i < len(groups); i++ {
-		group := strings.Replace(groups[i], "\\", "/", -1)
-		toReplace = strings.Replace(toReplace, "{"+strconv.Itoa(i)+"}", group, -1)
+		group := strings.ReplaceAll(groups[i], "\\", "/")
+		// Handling non-regular expression cases
+		for !isRegexp && !strings.Contains(toReplace, "{"+strconv.Itoa(placeHolderIndexer)+"}") {
+			placeHolderIndexer++
+			if placeHolderIndexer > maxPlaceholderIndex {
+				break
+			}
+		}
+		toReplace = strings.ReplaceAll(toReplace, "{"+strconv.Itoa(placeHolderIndexer)+"}", group)
+		placeHolderIndexer++
 	}
 	replaceOccurred := preReplaced != toReplace
-	return toReplace, replaceOccurred
+	return toReplace, replaceOccurred, nil
+}
+
+// Returns the higher index between all placeHolders target instances.
+// Example: for input "{1}{5}{3}" returns 5.
+func getMaxPlaceholderIndex(toReplace string) (int, error) {
+	reg := regexp.MustCompile(`\{(\d+?)}`)
```

Comment:
"MustCompile" panics if the expression cannot be parsed 
```suggestion
	reg, err := regexp.Compile(`\{(\d+?)}`)
	if err != nil {
		return 0, errorutils.CheckError(err)
	}
```